### PR TITLE
Add KMS provider

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

closes: https://github.com/elastic/cloudbeat/issues/704

this PR adds a KMS provider to fetch all symmetric keys and their rotation status, soon to be evaluated by rule 3.8. (KMS is also referred to in CIS AWS rule 3.7, but already implemented and doesn't require this provider)

### Screenshot/Data

- **Passed**
Screenshot 2023-02-28 at 7 57 52

- **Failed** 
Screenshot 2023-02-28 at 7 57 31

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

**Comment by orouz** [comment]
wasn't able to build: 

```
go build

vulnerability/runner.go:26:2: missing go.sum entry for module providing package github.com/aquasecurity/trivy/pkg/commands/artifact (imported by github.com/elastic/cloudbeat/vulnerability); to add:
	go get github.com/elastic/cloudbeat/vulnerability
vulnerability/runner.go:27:2: missing go.sum entry for module providing package github.com/aquasecurity/trivy/pkg/flag (imported by github.com/elastic/cloudbeat/vulnerability); to add:
	go get github.com/elastic/cloudbeat/vulnerability
vulnerability/scanner.go:34:2: missing go.sum entry for module providing package github.com/aquasecurity/trivy/pkg/types (imported by github.com/elastic/cloudbeat/vulnerability); to add:
	go get github.com/elastic/cloudbeat/vulnerability```

File: resources/providers/awslib/kms/provider.go
Comment (orouz) [inline resources/providers/awslib/kms/provider.go:67]: should the error log be prefixed with smth like `Failed to describe KMS key: %v`? 

it seems like all AWS sdk methods, including DescribKey, use [`invokeOperation`](https://github.com/aws/aws-sdk-go-v2/blob/main/service/kms/api_op_DescribeKey.go#L73-L86) and similarly they also return a string error, something like: "failed api operation for DescribeKey: %v` where %v is the actual error (like 401/400 http codes and a message)

File: resources/providers/awslib/kms/provider_test.go
Comment (orouz) [inline resources/providers/awslib/kms/provider_test.go:117]: Will add a test case for this once i figure out the Mock.On thingy. i'll remove this lines before merging/add a test.

File: go.mod
Comment (uri-weisman) [inline go.mod:35]: Why did you need to degrade Trivy?

**Review by uri-weisman — CHANGES_REQUESTED** [review]
Great work dude 💪 
Added some minor comments...

File: go.mod
Comment (orouz) [reply go.mod]: wasn't intentionally, happened after i ran go get for the kms package. after rebasing to latest `main` it's not an issue anymore 😄

File: resources/providers/awslib/kms/provider.go
Comment (orouz) [reply resources/providers/awslib/kms/provider.go]: changed to `k` to fit with `key` as each `KmsInfo` is a description of a key

File: resources/fetchers/kms_factory_test.go
Comment (orouz) [inline resources/fetchers/kms_factory_test.go:1]: code here is from `resources/fetchers/s3_factory_test.go`  with KMS changes.

File: resources/providers/awslib/kms/provider_test.go
Comment (orouz) [inline resources/providers/awslib/kms/provider_test.go:191]: these are used to ensure `DescribeKey` and `GetKeyRotationStatus` are both called with the right `KeyId` (using `mock.MatchedBy`) 

both do the same thing - check `KeyId` equals some string. it'd be nicer to make this a generic function

**Review by oren-zohar — APPROVED** [review]

**Comment by orouz** [comment]
@uri-weisman i think you need to approve after requesting changes otherwise merge is blocked

**Review by uri-weisman — APPROVED** [review]

**Pull request merged**
